src/Counter.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+type State = {
+  num: number
+};
+
+export class Counter extends React.Component<{}, State> {
+  state = {
+    num: 0,
+  };
+
+  addOne = () => {
+    this.setState(prevState => ({ num: prevState.num + 1 }));
+  };
+
+  add100 = () => {
+    this.setState(prevState => ({ num: prevState.num + 100 }));
+  };
+
+  increase = () => {
+    this.setState(prevState => {
+      if (!(prevState.num % 5)) {
```

Comment:
```suggestion
      if (prevState.num % 5 === 0) {
```
думаю так все таки надежнее